src/checker/text_import.cpp

```diff
@@ -61,27 +75,35 @@ struct text_importer {
         auto ls = read_level_params(in);
 
         inductive::inductive_decl decl(m_name.at(name_idx), ls, num_params, m_expr.at(type_idx), to_list(intros));
-        m_env = inductive::add_inductive(m_env, decl, true).first;
+        wrap_exception(decl.m_name, m_env, [&] { 
+            m_env = inductive::add_inductive(m_env, decl, true).first;
+        });
```

Comment:
Does it make sense to just do the wrapping in the `import_from_text` function rather than in the various places in this class?

Reply:
That's what I tried first, but then you don't get the name of the declaration.

Reply:
Makes sense, thanks.